content/posts/codegen.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Generating Code
+date: 2021-09-30
+draft: false
+summary: Let's talk about code generation!
+---
+
+# Generating Code
+
+At my day job at Google, I spend a lot of time generating code. Coming from
+outside of Google, this wasn't something I had ever thought about doing.
+Computers writing code for other computers to run has a very Skynet feel to it
+and sounded too complicated to ever be worth the effort.
+Yet, this practice is so commonplace that there's [built-in
+support](https://docs.bazel.build/versions/main/be/general.html#genrule) for it
+in Google's build system.
+
+Quick preface: I'm talking about writing a computer program that outputs
+some kind of file that's funny designed to be read in by a compiler or
+interpreter. I'm not talking about writing data files or schema files. It feels
+natural for computers to be writing those, since they broadly match up with the
+native data types found in programming languages.
```

Comment:
I think the grammar here is a little confusing. suggested rewrite for clarity (you don't have to take the way i rewrote it, you can do a different way too):
```suggestion
Quick preface: I'm talking about writing a computer program that outputs
some kind of funny file that's designed to be read by a compiler or
interpreter (not a human). I'm not talking about writing data files or schema files, which 
are pretty natural for computers to be writing, since they broadly match up with the
native data types found in programming languages.
```